### ISSUE TITLE
NixOS compatibility

### ISSUE THREAD

**Issue by dit7ya** [untitaker/quickenv#10]
```console

❯ quickenv shim croc
[ERROR quickenv] /home/halum/.quickenv/bin/croc is shadowed by an executable of the same name at /run/current-system/sw/bin/croc
```
I am wondering is quickenv fundamentally incompatible with Nix/NixOS. (direnv has first class support for nix).

Thanks for this project, btw!

**Comment by untitaker** [comment]
I'm not a nix user so I don't know what's going on. But I suppose nixos injects itself at the front of your path somehow? quickenv needs its own path to be (further) at the front in order to shim binaries.

**Comment by untitaker** [comment]
@dit7ya can you give some repro steps and dump your envvars here? I don't know what to do with this issue.

**Comment by dit7ya** [comment]
Sorry I don't exactly know where this error is coming from so I couldn't provide a shorter  repro step.

1. Get a Gitpod workspace for this repo - https://github.com/NixOS/nixpkgs/ . This gives you a nix installation super quickly.
2. Create a file called `default.nix` anywhere with the following content.
```nix

{ lib
, rustPlatform
, fetchFromGitHub
}:

rustPlatform.buildRustPackage rec {
  pname = "quickenv";
  version = "0.3.6";

  src = fetchFromGitHub {
    owner = "untitaker";
    repo = pname;
    rev = version;
    sha256 = "sha256-vXWVSoiYKPvmTtp8Rc9Wuf03KOpCJQ5EPAWUPN5FrvM=";
  };

  cargoSha256 = "sha256-WH7p6O74wXL0N0LOoUC61OD3DmLdSiwPZ66U3VJnKCc=";

  buildInputs = [ ];
  doCheck = false;

  meta = with lib; {
    description = "An unintrusive environment manager";
    homepage = "https://github.com/untitaker/quickenv";
    license = licenses.mit;
    maintainers = with maintainers; [ dit7ya ];
  };
}
```
This defines the build instructions for the nix package manager.

3. Run the following from the terminal to build the package. This needs to be run from the level where `default.nix` is.
`nix build --impure --expr 'with import (builtins.getFlake "nixpkgs") {}; pkgs.callPackage ./default.nix {}'`

4. Install the resulting binary to the system
`nix-env -i ./result`
5. Now `quickenv` should be in path.
6. Get into a `nix shell` which contains some package that we want to shim. For example
`nix shell nixpkgs#nodejs`. This enters you a shell where nodejs is installed.
7. Now try to run `quickenv shim node`. This is what fails.

Please let me know if anything is unclear.

**Comment by untitaker** [comment]
Okay, I think you're trying to package quickenv for nixos and the reason it fails is because you're not really installing quickenv fully. Like, you're installing the binary but you're not adding `~/.quickenv/bin/` to the user's PATH. It would be like installing direnv without the shell hook.

I followed your instructions in a docker container of nixos, so basically instead of gitpod i ran:

```
docker run -it nixos/nix
```

and when I get to run `quickenv shim node`, it errored exactly like you said:

```
[ERROR quickenv] /root/.quickenv/bin/node is shadowed by an executable of the same name at /nix/store/6yrgl6gc40938pmax3yd6xabl2mqmk70-nodejs-16.17.0/bin/node
```

Then I ran:

```
echo $PATH
```

and `/root/.quickenv/bin/` wasn't on there. That's the problem.
